neps_examples/convenience/neps_tblogger_tutorial.py

```diff
@@ -0,0 +1,373 @@
+"""
+NePS tblogger With TensorBoard
+====================================
+1- Introduction
+---------------
+
+Welcome to the NePS tblogger with TensorBoard tutorial. This guide will walk you
+through the process of using the NePS tblogger class to effectively monitor and
+analyze performance data from various model configurations during training.
+
+Assuming you already have experience in NePS, the main reason of creating this tutorial is to showcase the
+power of visualization using tblogger. if you wish to directly reach that part, check the lines
+between 244-264 or search for 'Start Tensorboard Logging'
+
+2- Learning Objectives
+----------------------
+
+By completing this tutorial, you will:
+
+- Understand the role of NePS tblogger and its importance in HPO and NAS.
+- Learn how to define search spaces within NePS to explore different model configurations.
+- Build a comprehensive run pipeline to train and evaluate models.
+- Utilize TensorBoard to visualize and compare performance metrics of different model configurations.
+
+3- Setup
+--------
+
+Before we dive in, make sure you have the necessary dependencies installed. If you haven't already,
+install the ``NePS`` package using the following command:
+
+```bash
+pip install neural-pipeline-search
+```
+
+Additionally, please note that NePS does not include ``torchvision`` as a dependency.
+You can install it with the following command:
+
+```bash
+pip install torchvision==0.14.1
+```
+
+These dependencies will ensure you have everything you need to follow along with this tutorial successfully.
+"""
+
+import argparse
+import logging
+import os
+import random
+import shutil
+import time
+from typing import Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchvision
+from torch.optim import lr_scheduler
+from torch.utils.data.dataloader import DataLoader
+from torch.utils.data.sampler import SubsetRandomSampler
+from torchvision.transforms import transforms
+
+import neps
+from neps.plot.tensorboard_eval import tblogger
```

Comment:
what happens if I import and have forgotten to do `tblogger.add_scalar(...)` and have also not done `tblogger.disable()`? 
Or does a new import of blogger in a local scope resets the state?

Can we have something like `tblogger.record_status=True/False` which the user can print to verify and debug?
I am thinking along the lines of `torch.cuda.is_available()`. 

Also, looking at the example, we should perhaps write somewhere that by default the flag to record is set to True.

Reply:
tblogger now has tblogger.get_status() this returns true when logger is enabled and false when it is disabled to keep track to tblogger's state.